Sources/SourceKitLSP/Swift/SwiftLanguageServer.swift

```diff
@@ -306,6 +379,7 @@ extension SwiftLanguageServer {
     }
     self.publishDiagnostics(
         response: dict, for: snapshot, compileCommand: compileCmd)
+    self.updateLexicalAndSyntacticTokens(response: dict, for: snapshot)
```

Comment:
Would it be worth transmitting the new syntax information to the client here already instead of waiting for semantic highlighting? Otherwise, we won’t have any SourceKit-LSP-provided highlighting in case `sourcekitd` crashes (which pauses semantic requests for 10-20 seconds but still provides syntactic functionality) or if the semantic requests take too long for some other reason.

Reply:
Yes, that would however complicate testing a fair bit AFAICT. Right now, every refresh request indicates that the server knows all tokens, both semantic and syntactic. Since the refresh request doesn't state which kind of tokens the server has computed, we would have to make fragile assumptions, e.g. that the server sends exactly 2 refresh requests (syntactic + semantic) until we get the right tokens. But since syntactic tokens are updated in several places, it would be easy to introduce more refresh requests that accidentally break these tests.

Reply:
Also note that clients like VSCode request tokens whenever they want to, e.g. after opening or editing a document, not first after the refresh request, so the highlighting should be responsive in any case.

Reply:
We'll leave it like this for now, see the discussion above.